pkg/controllers/cloud_config_sync_controller.go

```diff
@@ -218,10 +191,11 @@ func (r *CloudConfigReconciler) prepareSourceConfigMap(source *corev1.ConfigMap,
 		delete(cloudConfCm.Data, infraConfigKey)
 		return cloudConfCm, nil
 	}
-	return nil, fmt.Errorf(
-		"key %s specified in infra resource does not found in source configmap %s",
-		infraConfigKey, client.ObjectKeyFromObject(source),
-	)
+
+	// If there was no relevant data in the configmap, make a default entry for it.
+	cloudConfCm.Data[defaultConfigKey] = ""
+
+	return cloudConfCm, nil
```

Comment:
Not sure about this one. If the user has specced a custom key and that custom key isn't there, we probably still want to error

I assume our fallback detects an that neither the managed or unmanaged confs exist, so creates a blank entry which exists this function on L185? So this change shouldn't be needed?